### DISCUSSION TITLE
calling in data variable from previous trial (dynamic variable?)

### DISCUSSION THREAD

**Discussion by csavasegal** [jspsych/jsPsych#2336] [Q&A]
Hello,

I've been trying to figure out how to reference a variable from a previous trial in an upcoming trial and have not been able to figure out how to and none of the solutions for similar problems have worked for me. 

I have a trial that is a survey html form which saves out three things in the response variable called : first, second, third. I then reference these in the on_finish function of this trial as so: 
```
    on_finish: function(data){
        window.alert(data.response.first);
        var self_ML = questions[k].ML1 + data.response.first;
        window.alert(self_ML);
        // data.self_ML = self_ML
        // jsPsych.data.addProperties({subject_: self_ML})
    }

}
```
This works out well when I get my data at the end of the experiment: 

```
"response": {
			"first": "a",
			"second": "aa",
			"third": "aaa"
		},
		"trial_type": "survey-html-form",
		"trial_index": 0,
		"time_elapsed": 3068,
		"internal_node_id": "0.0-0.0-0.0",
		"participant": 37,
		"self_ML": "This small child is a"
```

However, I cannot figure out how to reference the variable in future trials. I am trying to show it on a slider eventually. This is the current code that I am trying to use: 
```
timeline: [ {
        type: "image-slider-response",
        trial_duration: 5000,
        stimulus: questions[k].img,
        labels: [JSON.stringify(jsPsych.data.get().select('self_ML')), questions[k].ML2],
        require_movement: true,
        post_trial_gap: 1000,
},}
]
}

```

 It says that the object (when converted to string) is empty. Like this: {"values":[]}. I put self_ML in both an on_finish and an on_trial_finish and neither worked too. I'd really appreciate any insight anyone has. Thanks!! 

Ideally, I would add this variable as a timelineVariable but that seems impossible? 
I am using jsPsych6.1 to integrate with pavlovia by the way.

**Comment by jodeleeuw** [comment]
I'm going to convert this to a discussion.

**Reply by csavasegal** [reply]
Figured it out! The solution was to define the variable within a function rather than just as a variable. Posting it below in case anyone else tries to do something similar:

```
labels: function(data){
          temp = jsPsych.data.get().last(1).values()[0].YOURVARIABLE;
          return [temp, temp]
        }
```

Sorry I posted the wrong spot!
